### PR TITLE
Fix(code): Message Log, check if a message to compare exists

### PR DESCRIPTION
**Pull request by RisingLeaf**
**Bug fix**

This PR addresses the bug/feature described in issue #9674

## Summary
Adds a missing check otherwise you get a nullptr

## Testing Done
Launched on the same type of machine as in the issue, but that should not be related

### CODE REVIEW

**Comment by bene-dictator** [comment]
It works. 👍

**Comment by TomGoodIdea** [comment]
> Launched on the same type of machine as in the issue, but that should not be related

Tested master on Linux and Windows (unfortunately I don't have access to a Silicon machine) and got no segfaults, so this must be a platform-dependent bug.

File: source/Messages.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ void Messages::Add(const string &message, Importance importance)
 {
 	lock_guard<mutex> lock(incomingMutex);
 	incoming.emplace_back(message, importance);
-	if(message != logged.front().first)
+	if(!logged.size() || message != logged.front().first)
```
Comment (TomGoodIdea) [inline source/Messages.cpp:42]:
```suggestion
	if(logged.empty() || message != logged.front().first)
```

Comment (RisingLeaf) [reply source/Messages.cpp]: good catch

**Comment by warp-core** [comment]
> > Launched on the same type of machine as in the issue, but that should not be related
> 
> Tested master on Linux and Windows (unfortunately I don't have access to a Silicon machine) and got no segfaults, so this must be a platform-dependent bug.

I can reproduce the crash on Windows with the Microsoft Visual C++ runtime (after building with clang). It actually has a debug assertion for calling front() on an empty vector. Building with gcc (either in CodeBlocks or VS Code) and the libraries it provides results in no apparent problem.

**Pull request merged**